fish-pepper/run-java-sh/fp-files/java-default-options

```diff
@@ -8,47 +8,41 @@
 # Env Vars evaluated:
 
 # JAVA_OPTIONS: Checked for already set options
-# JAVA_MAX_MEM_RATIO: Ratio use to calculate a default maximum Memory, in percent.
-#                     E.g. the default value "50" implies that 50% of the Memory
+# JAVA_MAX_MEM_RATIO: Ratio use to calculate a maximum Memory, in percent.
+#                     E.g. the value "50" implies that 50% of the Memory
 #                     given to the container is used as the maximum heap memory with
-#                     '-Xmx'. It is a heuristic and should be better backed up with real
-#                     experiments and measurements.
+#                     '-Xmx'.
 #                     For a good overviews what tuning options are available -->
 #                             https://youtu.be/Vt4G-pHXfs4
 #                             https://www.youtube.com/watch?v=w1rZOY5gbvk
 #                             https://vimeo.com/album/4133413/video/181900266
-# Also note that heap is only a small portion of the memory used by a JVM. There are lot
-# of other memory areas (metadata, thread, code cache, ...) which addes to the overall
-# size. There is no easy solution for this, 50% seems to be are reasonable compromise.
-# However, when your container gets killed because of an OOM, then you should tune
-# the absolute values
-#
+# Note that heap is only a small portion of the memory used by a JVM. There are lot
+# of other memory areas (metadata, thread, code cache, ...) which add to the overall
+# size. When your container gets killed because of an OOM, then you should tune
+# the absolute values.
 
 # Check for memory options and calculate a sane default if not given
-max_memory() {
-  # Check whether -Xmx is already given in JAVA_OPTIONS. Then we dont
-  # do anything here
+heap_size() {
+  # Check whether -Xmx is already given in JAVA_OPTIONS
   if echo "${JAVA_OPTIONS}" | grep -q -- "-Xmx"; then
     return
   fi
 
-  # Check if explicitly disabled
-  if [ "x$JAVA_MAX_MEM_RATIO" = "x0" ]; then
-    return
-  fi
-
-  # Check for the 'real memory size' and caluclate mx from a ratio
-  # given (default is 50%)
-  if [ "x$CONTAINER_MAX_MEMORY" != x ]; then
+  # Check for the 'real memory size' and calculate Xmx from the ratio
+  if [ "x$CONTAINER_MAX_MEMORY" != x -a "x$JAVA_MAX_MEM_RATIO" != x ]; then
     local max_mem="${CONTAINER_MAX_MEMORY}"
-    local ratio=${JAVA_MAX_MEM_RATIO:-50}
+    local ratio="${JAVA_MAX_MEM_RATIO}"
     local mx=$(echo "${max_mem} ${ratio} 1048576" | awk '{printf "%d\n" , ($1*$2)/(100*$3) + 0.5}')
     echo "-Xmx${mx}m"
   fi
 }
 
+memory_options() {
+  echo "-XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap $(heap_size)"
```

Comment:
Are these options also available on Java 8 ? Up to now the run script is not bound a specific Java version. 

The same question is of course true also for the more advanced tuning options (like in the diagnostics options)

Reply:
These options are available starting JDK 8u131+. I made the assumption it's a satisfied requirement based on the Java and s2i images but forgot the run script can be used elsewhere.

If we decide to keep the 50% heuristic when memory limit are greater than a minimum threshold as discussed in #31, then it may be OK to get rid of these options.